astropy/coordinates/baseframe.py

```diff
@@ -134,6 +137,214 @@ class RepresentationMapping(NamedTuple):
+
+        attrs.extend(frame_transform_graph.frame_attributes.keys())
+
+        out = super()._represent_as_dict(attrs)
+
+        out["representation_type"] = coord.representation_type.get_name()
+        # Note that coord.info.unit is a fake composite unit (e.g. 'deg,deg,None'
+        # or None,None,m) and is not stored. The individual attributes have
+        # units.
+
+        return out
+
+    def new_like(self, coords, length, metadata_conflicts="warn", name=None):
+        """A new consistent coordinate instance with the given length.
+
+        Return a new SkyCoord or BaseCoordinateFrame instance which is
+        consistent with the input coordinate objects ``coords`` and has
+        ``length`` rows.  Being "consistent" is defined as being able to set an
+        item from one to each of the rest without any exception being raised.
+
+        This is intended for creating a new coordinate instance whose elements
+        can be set in-place for table operations like join or vstack.  This is
+        used when a coordinate object is used as a mixin column in an astropy
+        Table.
+
+        The data values are not predictable and it is expected that the consumer
+        of the object will fill in all values.
+
+        Parameters
+        ----------
+        coords : list
+            List of input SkyCoord or BaseCoordinateFrame objects
+        length : int
+            Length of the output SkyCoord or BaseCoordinateFrame object
+        metadata_conflicts : str ('warn'|'error'|'silent')
+            How to handle metadata conflicts
+        name : str
+            Output name (sets output coord.info.name)
+
+        Returns
+        -------
+        coord : |SkyCoord|, |BaseCoordinateFrame| (or subclass)
+            Instance of this class consistent with ``coords``
+
+        """
+        # Get merged info attributes like shape, dtype, format, description, etc.
+        attrs = self.merge_cols_attributes(
+            coords, metadata_conflicts, name, ("meta", "description")
+        )
+        coord0 = coords[0]
+
+        # Make a new coord object with the desired length and attributes
+        # by using the _apply / __getitem__ machinery to effectively return
+        # coord0[[0, 0, ..., 0, 0]]. This will have the all the right frame
+        # attributes with the right shape.
+        indexes = np.zeros(length, dtype=np.int64)
+        out = coord0[indexes]
+
+        # Use __setitem__ machinery to check for consistency of all coords
+        for coord in coords[1:]:
+            try:
+                out[0] = coord[0]
+            except Exception as err:
+                raise ValueError("Input coords are inconsistent.") from err
+
+        # Set (merged) info attributes
+        for attr in ("name", "meta", "description"):
+            if attr in attrs:
+                setattr(out.info, attr, attrs[attr])
+
+        return out
+
+    def _insert(self, obj, values, axis=0):
```

Comment:
This is essentially the old `SkyCoord.insert` method - this avoids defining effectively the same method twice. I considered defining a mixin class, but since this method uses `_new_like`, it seemed reasonable to define it on the info class.